src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PeopleRow } from '../PersonRow';
+import { useMemo } from 'react';
+import { SearchParamsForFilter } from '../../types/SearchParamsForFilter';
+import { SortParams } from '../../types/SortParams';
+
+type Props = {
+  people: Person[];
+};
+
+function filterPeopleBySearchParams(
+  peopleList: Person[],
+  { queryParam, centuriesParam, peopleSexParam }: SearchParamsForFilter,
+) {
+  const queryInLowerCase = queryParam.toLowerCase();
+
+  return peopleList.filter(person => {
+    const personCentury = (+person.born.toString().slice(0, 2) + 1).toString();
```

Comment:
Your approach looks really weird and doesn't handle cases where the birth year is at the start of a new century. You can use something simpler like:
const personCentury= Math.ceil(person.born / 100).toString();

Reply:
looks better